### PR TITLE
feat(integration tests): Utilities for testing ansible-trace

### PR DESCRIPTION
**Pull request by louisquentinjoucla**
Hello everyone :wave: !

This is the first step for #10 

This proposition allow creating tests using pytest framework where we can chose strategy, inventory for the same playbook file (see tests/integration/tests/basic.py)
I have also made some utilities functions that parse trace to ensure its integrity.
This include also related github action CI.yml (here a [sample](https://github.com/louisquentinjoucla/ansible-trace/actions/runs/2482547964))

Waiting for your feedback,
Thanks!

### CODE REVIEW

File: tests/integration/inventories/multiple_hosts.ini [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[all]
+127.0.0.1 ansible_connection=local
```
Comment (mhansen) [inline tests/integration/inventories/multiple_hosts.ini:2]: This is pretty clever, using multiple localhosts. Nice idea

File: tests/integration/tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import pytest
+from fixtures import ansible_play
+
```
Comment (mhansen) [inline tests/integration/tests/conftest.py:3]: What's this file? Probably missing something but is it empty? Shouldn't there be, like, a test function or something in here? (I'm not very familiar with pytest)

File: tests/integration/tests/fixtures.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+def ansible_play(request):
+   
+    # Get required marks, raise exception if not defined or invalid params 
+    inventory = request.node.get_closest_marker('ansible_inventory').args[0]
+    strategy = request.node.get_closest_marker('ansible_strategy').args[0]
+    playbook = request.node.get_closest_marker('ansible_playbook').args[0]
+
+    # Set params for ansible runner
+    request.config.option.ansible_playbook_directory = "."
+    request.config.option.ansible_playbook_inventory = "inventory.ini"
```
Comment (mhansen) [inline tests/integration/tests/fixtures.py:15]: Is there a reason we can't do `= inventory` ? Wouldn't that let us avoid the symlinking and renaming (and the resulting side-effects on the filesystem?)

File: tests/integration/tests/utils.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+    event = HostEvent(event_hash)
+
+    hosts[event.pid] = event
+    duration_events[event.pid] = {}
+    duration_stacks[event.pid] = deque()
+
+    return (hosts, duration_events, duration_stacks)
+
+def add_duration_event(hosts: Dict[int, HostEvent], events: Dict[int, Any], curr_stacks: Dict[int, Deque], event_hash: Any) -> Tuple[Dict[int, Any], Dict[int, Deque]]:
```
Comment (mhansen) [inline tests/integration/tests/utils.py:46]: Would you mind wrapping some of these very long lines? Whatever the other python files in the repo are wrapped at

File: tests/integration/tests/basic.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+from event import HostEvent
+
+@pytest.mark.ansible_playbook('basic/basic.yml')
+@pytest.mark.ansible_inventory('inventories/multiple_hosts.ini')
+@pytest.mark.ansible_strategy('free')
+def test_basic_multiple_free(ansible_play):
+    trace_hosts: Dict[int, HostEvent]
+    trace_events: Dict[int, Any]
+    trace_json: JSONTYPE = get_last_trace()
+    trace_hosts, trace_events = parse_trace(trace_json)
```
Comment (mhansen) [inline tests/integration/tests/basic.py:16]: This last line feels a little funny; we're assigning variables we aren't using. I had to read the `parse_trace` to figure out that the bulk of the test is in there: that it's validating the trace as it's parsing. Could we name the function to make clear it's validating too? `parse_and_validate_trace` or something?

**Comment by mhansen** [comment]
This overall looks really good to me. Thanks so much for writing these tests. I really like how they're integration-level tests that actually run Ansible, and have semantic validations on the output (e.g. timestamps nest).

File: tests/integration/tests/conftest.py (hunk 2 above)
Comment (louisquentinjoucla) [reply tests/integration/tests/conftest.py:3]: It is to import the fixture on every other pytest test file

File: tests/integration/tests/fixtures.py (hunk 3 above)
Comment (louisquentinjoucla) [reply tests/integration/tests/fixtures.py:15]: Oops my bad, forgot to unstash some of my work, will definitely fix that

File: tests/integration/tests/utils.py (hunk 4 above)
Comment (louisquentinjoucla) [reply tests/integration/tests/utils.py:46]: I will

**Comment by louisquentinjoucla** [comment]
Review taken in account ! Lint pep8, removed the symlinks stuff, and changed sub name. 
Waiting for this one to be merge before rebasing other PRs!

Do not hesitate if something is still not good!

**Review by mhansen — APPROVED** [review]
Looks great, thanks, and thanks for cleaning up my code too

**Pull request merged**